packages/utils/src/options.ts

```diff
@@ -1,64 +1,77 @@
+import { Provider } from "@tko/provider";
+
+interface CustomBindingGlobalProperties {
+  String;
+  isObservable;
+}
+
+
 //
 // This becomes ko.options
 // --
 //
 // This is the root 'options', which must be extended by others.
+export class Options {
+  [key: string]: any;
+  
+  deferUpdates: boolean = false
 
-const options = {
-  deferUpdates: false,
+  useOnlyNativeEvents: boolean = false
 
-  useOnlyNativeEvents: false,
-
-  protoProperty: '__ko_proto__',
+  protoProperty: string = '__ko_proto__'
 
     // Modify the default attribute from `data-bind`.
-  defaultBindingAttribute: 'data-bind',
+  defaultBindingAttribute: string = 'data-bind'
 
     // Enable/disable <!-- ko binding: ... -> style bindings
-  allowVirtualElements: true,
+  allowVirtualElements: boolean = true
+
 
     // Global variables that can be accessed from bindings.
-  bindingGlobals: Object.create(null),
+  bindingGlobals: Object & CustomBindingGlobalProperties = Object.create(null)
 
     // An instance of the binding provider.
-  bindingProviderInstance: null,
+  bindingProviderInstance: Provider
 
   // Whether the `with` binding creates a child context when used with `as`.
-  createChildContextWithAs: false,
+  createChildContextWithAs: boolean = false
 
     // jQuery will be automatically set to globalThis.jQuery in applyBindings
     // if it is (strictly equal to) undefined.  Set it to false or null to
     // disable automatically setting jQuery.
-  jQuery: globalThis.jQuery,
+  jQuery : JQueryStatic | boolean | null = (globalThis as any).jQuery
 
-  Promise: globalThis.Promise,
+  Promise: PromiseConstructor = globalThis.Promise
 
-  taskScheduler: null,
+  taskScheduler: any = null
 
-  debug: false,
+  debug: boolean = false
 
-  global: globalThis,
-  document: globalThis.document,
+  global: any = globalThis
+  document: Document = globalThis.document
 
     // Filters for bindings
     //   data-bind="expression | filter_1 | filter_2"
-  filters: {},
+  filters: any = {}
 
   // Used by the template binding.
-  includeDestroyed: false,
-  foreachHidesDestroyed: false,
+  includeDestroyed: boolean = false
 
-  onError: function (e) { throw e },
+  foreachHidesDestroyed: boolean = false
 
-  set: function (name, value) {
-    options[name] = value
-  },
+  onError (e : Error) : void { throw e }
```

Comment:
stylistically, the `x:type` is preferable to `x : type`, the latter looking a bit too much like a ternary. (not a blocker, just noting)